site/content/docs/main/release-instructions.md

```diff
@@ -1,16 +1,103 @@
 ---
 title: "Release Instructions"
 layout: docs
+toc: "true"
 ---
-
 This page covers the steps to perform when releasing a new version of Velero.
 
-## Preparing for a release
-
-### (GA Only) Create release blog post
-
-For each major or minor release, create and publish a blog post to let folks know what's new.
-
+## General notes
+- Please read the documented variables in each scrip to understand what they are for and how to properly format their values.
+- You will need to have a remote named `origin` pointing to the Velero repository so the scripts can work properly.
+- GA release: major and minor releases only. Example: 1.0 (major), 1.5 (minor).
+- Pre-releases: Any release leading up to a GA. Example: 1.4.0-beta.1, 1.5.0-rc.1
+- RC releases: Release Candidate, contains everything that is supposed to ship with the GA release. This is still a pre-release.
+
+## Preparing
+
+### Create release blog post (GA only)
+For each major or minor release, create and publish a blog post to let folks know what's new. Please follow these [instructions](Instructions-to-write-and-release-a-blog-post).
+
+### Changelog and Docs PR
+#### Troubleshooting
+- `You don't have enough free space in /var/cache/apt/archives/` when running `make serve-docs`: run `docker system prune`.
+
+#### Steps
+1.  If it doesn't already exist: in a branch, create the file `changelogs/CHANGELOG-<major>.<minor>.md` by copying the most recent one.
+2.  Update the file `changelogs/CHANGELOG-<major>.<minor>.md`
+	- Run `make changelog` to generate a list of all unreleased changes.
+    - Copy/paste the output into `CHANGELOG-<major>.<minor>.md`, under the "All Changes" section for the release. 
+	- You *may* choose to tweak formatting on the list of changes by adding code blocks, etc.
+	- 	Update links at the top of the file to point to the new release version
+3.  Update the main `CHANGELOG.md` file to properly reference the release-specific changelog file
+	- Under "Current release": 
+	    - Should contain only the current GA release.
+    - Under "Development release": 
+	    - Should contain only the latest pre-release
+	    - Move any prior pre-release into "Older releases"
+4. GA Only: Remove all changelog files from `changelogs/unreleased`.
+5. Generate new docs
+	- Run `make gen-docs`, passing the appropriate variables. Examples:
+		a) `VELERO_VERSION=v1.5.0-rc.1 NEW_DOCS_VERSION=v1.5.0-rc.1 make gen-docs`.
+		b) `VELERO_VERSION=v1.5.0 NEW_DOCS_VERSION=v1.5 make gen-docs`).
```

Comment:
For a standard GA release, should the new docs version always omit the patch component of the version?

Reply:
Yes. We used to do patch release-level docs, but we now only do minor release-level (so v1.5) docs in order to make getting patch releases out easier.

Reply:
This is not what I did for the RC (see it is v1.5.0-rc.1). The instructions made me think all non GA docs should be under `-pre`. We need to decide. I think it should not be `-pre` (see this PR description. Options are:
- v1.5.0-rc.1 (full version)
- v1.5 (omit the patch)

Reply:
For a GA release, which is what @zubron was asking about, no patch number at the end.

For non-GA releases, yes, putting them under `-pre` was what was happening before. I like keeping `pre` because for those, we're iterating and can/should be deleting those docs directories anyway, so we could modify the scripts to clean up an existing `<version>-pre` directory. Pre-release version docs, at least in my mind, are largely ephemeral, and will be gone once the GA release comes out.

Using the explicit version number means we're generating dedicated directories per pre-release version, and I don't really see the benefit to having those stick around.

Reply:
What would be the difference between using `<version>-pre` vs something like `v1.5.0-rc.1`?

Both would be deleted once GA is out. `<version>-pre` to me is just one more term to wrap our head around.

Reply:
`-pre` covers all pre-releases, so alpha, beta, and RC releases.

Reply:
> -pre covers all pre-releases, so alpha, beta, and RC releases.

How do you do the copy/paste if the origin and target folders are named the same?

Reply:
The use can set the `PREVIOUS_DOCS_VERSION` value themselves (https://github.com/vmware-tanzu/velero/blob/main/hack/gen-docs.sh#L37), but I think I see your point now, I was missing it before :)

This is surfacing a bug in the behavior of the `gen-docs.sh` script, which assumes that the latest release is always the last major version and you won't have multiple pre-releases.

In which case we either a) need to fix the script to account for this or b) update the instructions to explicitly set the `PREVIOUS_DOCS_VERSION` or some third option I'm not thinking of...

Reply:
But what do you set `PREVIOUS_DOCS_VERSION` to, this is my question.

Example: 
1) We have `v1.6.0-beta.1` release
2) The docs for that is under `v1.6-pre`
3) We want to release `v1.6-beta.2`
3) What should the `PREVIOUS_DOCS_VERSION` be? If we say `v1.6-pre`, then it can't copy itself to `v1.6-pre` right?

Reply:
Yeah, I edited my comment, but I think it should be `v1.5` for all pre-release versions in your example. You're correct that you couldn't do `v1.6-pre` for both unless we did some extra move/swap logic, which doesn't seem worthwhile to add. 

That would match the intended logic of latest, non-pre-release version that the script is expecting, while getting around the fact that the real latest based on sorting would be `v1.6-pre`.

Reply:
To answer your question directly, here's what I envision.

For beta.1:

`PREVIOUS_DOCS_VERSION=v1.5`
`NEW_DOCS_VERSION=v1.6-pre`

All files in the `v1.6-pre` directory will be newly created.

For beta.2:

`PREVIOUS_DOCS_VERSION=v1.5`
`NEW_DOCS_VERSION=v1.6-pre`

All files in the `v1.6-pre` directory will be overwritten, only those with changes should show up in a `git status` though. It's totally plausible that the docs don't even change.

Reply:
> PREVIOUS_DOCS_VERSION=v1.5
NEW_DOCS_VERSION=v1.6-pre

> All files in the v1.6-pre directory will be overwritten

With the docs from `v1.5`, right?

So what happens if during `beta.1` we had updates to the docs? Wouldn't they get lost? What am I missing?

Reply:
Good point... we'd actually want `main`, since that's where new docs should go. We shouldn't be putting new docs into the pre-release directories.

Let me actually run the script and try it.

Reply:
`main` works as the source, but without some changes to the gen-docs script, it'll block on using the same source directory since https://github.com/vmware-tanzu/velero/blob/main/hack/gen-docs.sh#L30 fails if the directory exists.

I'm inclined to defer this discussion for a separate PR that would make the necessary changes OR just use the versions directly, but we moved to the `-pre` format for a reason; regenerating docs every time we had a beta was cumbersome. Likewise with the patch-level docs - they were cumbersome and took up space for little benefit.

Reply:
Ok. I guess that's why the original instructions says to delete the `$NEW_DOCS_VERSION

![image](https://user-images.githubusercontent.com/16508/93274772-f90c8800-f76f-11ea-9f98-0da681afc775.png)
![image](https://user-images.githubusercontent.com/16508/93274894-4852b880-f770-11ea-8b1c-578ca982d527.png)

But it also says to run `make gen-docs` for pre-releases because at this point the folder and meta data have been deleted, so I think regenerating the docs will be necessary no matter what.

If I'm right, then we are back at my question: what difference does it make to wipe and recreate a `-pre` directory vs using the entire pre-release name (Ex `1.5.0-rc.1`), and wiping that out and using the GA name (Ex `1.5.0`)?

Reply:
I think the difference is semantics at this point, since the instructions say to remove the pre-release stuff each time anyway (we should automate it). Doing the full pre-release version string is more precise, but deviates from how GA works, which leaves off the patch number now (as of v1.4).

Since we're deleting it as we go, I'm not sure I hold a strong opinion either way. For now, I think we can defer the discussion and I'll review the rest of the PR.